Attack hook functionality and sound muting

NovaAstral opened the pull request:
- Added the ability for the Primary/Secondary attack functions to be used
- Added the ability to mute a sonics sound by setting self.mutesonic to true on the client

NovaAstral committed:
Attack hook functionality and sound muting
- Added the ability for the Primary/Secondary attack functions to be used
- Added the ability to mute a sonics sound by setting self.mutesonic to true on the client

NovaAstral committed:
stop sounds from being stopped if sonic is muted
-Stopped the sonic from stopping the sound if you emitsound with the same file path as one the sonicsd you're holding uses

NovaAstral committed:
remove accidental newline

AmyJeanes reviewed (changes requested):
Can we swap out use of `x == true` to just `x` and `x == false` to `not x`? Otherwise looks good!

NovaAstral committed:
switch true/false to var/notvar and add disablereload
-Switched true / false checks to var / not var checks
-Added the ability to disable the 'Call TARDIS' Reload function of the sonicsd with self.disablereload

AmyJeanes commented on lua/weapons/swep_sonicsd/modules/cl_sound.lua:
```diff
@@ -16,6 +16,7 @@ end
 SWEP:AddHook("Initialize", "sound", function(self)
     self.curbeep=0
     self.eyeangles=Angle(0,0,0)
+    self.mutesonic=false
```
Wondering if it might be better to do hooks for this and the prevent reload thing too 

like `if self:CallHook("CanReload") == false then return end` in the main reload code or something and similar for muting sonic like `ShouldPlaySounds`

Note that I'm allowing and encouraging `== false` here because it's different to returning `nil` or not returning anything

NovaAstral replied:
It probably would be better, but I personally don't actually know how to go about doing that, which is why its the way it is right now

AmyJeanes replied:
Well that is how you do it 😁 gimme a shout on Discord if you get stuck
